r/geoarrow/R/wk-compat.R

```diff
@@ -112,6 +112,60 @@ infer_nanoarrow_schema.wk_xy <- function(x, ...) {
   wk_geoarrow_schema(x, na_extension_geoarrow, "POINT", dimensions = dims)
 }
 
+#' @export
+convert_array.wk_wkt <- function(array, to, ...) {
+  schema <- infer_nanoarrow_schema(array)
+  geo <- geoarrow_schema_parse(schema)
+  vctr <- as_geoarrow_vctr(array)
+
+  if (geo$extension_name == "geoarrow.wkt") {
```

Comment:
For a future pr, I think we could extract some of this into `as_wkt`, `as_wkb`, `as_xy` methods.

Reply:
Then these could be skeletons. Or the reverse, and as_ methods are just calling these methods.